### PR TITLE
Fixes and updates in RecSys'21 paper reproducibility

### PR DESCRIPTION
**Pull request by gabrielspmoreira**
Fixed reproducibility commands and added a link to the [new scripts](https://github.com/NVIDIA-Merlin/Transformers4Rec/pull/289) using the released Transformers4Rec PyTorch API

### CODE REVIEW

File: 2021_acm_recsys_transformers4rec/README.md [hunk 1]
```diff
@@ -18,13 +18,6 @@ In order to demonstrate the usefulness of the library for research and also to v
...
-
-### Training and evaluation
-The command lines to run each experiment group (dataset and algorithm) reported in the paper can be found [here](experiments_reproducibility_commands.md), with the corresponding best hyperparameters reported in [Appendix C](Appendices/Appendix_C-Hyperparameters.md).
-
-**IMPORTANT**: It is worthwhile to mention that since our experiments for the paper were finished the library was completely refactored by NVIDIA Merlin team, converting it from a research library into an flexible [open-source library](https://github.com/NVIDIA-Merlin/Transformers4Rec), packaged with PyTorch and TensorFlow APIs. We are working on preparing Python scripts to reproduce our paper results with the refactored version of Transformers4Rec and will release those scripts soon in this [NVIDIA-Merlin/publications](https://github.com/NVIDIA-Merlin/publications/tree/main/2021_acm_recsys_transformers4rec) repo, so that you can more easily understand and extend the library for your own research on sequential and session-based recommendation ;)
```
Comment (rnyak) [inline 2021_acm_recsys_transformers4rec/README.md:30]: I am a bit confused. Arent you  preparing the  scripts for paper repro on the main branch of transformers4rec GH repo? Here it gives different link. It refers to Publications repo..

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md [hunk 2]
```diff
@@ -1,4 +1,16 @@
-# Experiments Reproducibility Instructions
+# Experiments reproducibility
+The experiments for the Transformers4Rec paper were performed in a former version of the Transformers4Rec library tagged as [recsys2021](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/recsys2021), which can be used for reproducibility. This document provides instruction for reproducing the experiments with that original (pre-release) codebase we used for the Transformers4Rec paper. 
+
+**IMPORTANT**: For researchers and practioners aiming to perform experiments similar to the ones presented in our paper (e.g. incremental training and evaluation of session-based recommendation with Transformers), we strongly encourage the usage of the [latest version](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples/t4rec_paper_experiments) of the experiment scripts which were updated to use the released PyTorch API, because it was completely refactored, is more modularized and documented than the original scripts, and is supported by the NVIDIA Merlin team.
```
Comment (rnyak) [inline 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: better to say `well documented than the original scripts, ...`

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md [hunk 3]
```diff
@@ -11,8 +23,11 @@ DATA_ROOT_PATH=~/transformers4rec_paper_preproc_datasets_public
 3) Go to the project root path: `cd Transformers4Rec/`
 4) Create a `conda` environment and install the library dependencies, according to the instructions [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/recsys2021/hf4rec)
 4) Run the command of one of the following experiment groups (dataset, algorithm) to reproduce paper results. P.s. The reported numbers in the paper are the average of the metrics of 5 runs with different random seeds (`--seed`).
```
Comment (rnyak) [inline 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: I will switch the order of 3) and 4) and then add additional command in the 4):

3) Create a `conda` environment and install the library dependencies, according to the instructions [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/recsys2021/hf4rec)
4) Go to the project root path: `cd Transformers4Rec/` and run `pip install -r requirements.txt`

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md [hunk 4]
```diff
@@ -11,8 +23,11 @@ DATA_ROOT_PATH=~/transformers4rec_paper_preproc_datasets_public
 3) Go to the project root path: `cd Transformers4Rec/`
 4) Create a `conda` environment and install the library dependencies, according to the instructions [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/recsys2021/hf4rec)
 4) Run the command of one of the following experiment groups (dataset, algorithm) to reproduce paper results. P.s. The reported numbers in the paper are the average of the metrics of 5 runs with different random seeds (`--seed`).
+5) When you run the commands, it will be generated within `--output_dir` CSV files with the metrics for each evaluation time window and also the Average Over Time (AOT) metrics reported in the paper.  
+6) You can chech the metrics plots with Tensorboard (logs saved in the output dir) and with [Weights & Biases](http://wandb.ai/) service (with a free account). For that you need to install wandb (`pip install wandb`) and run `wandb login` to provide your authentication key. The run stats and plots will be saved to the `huggingface` project on W&B service by default.
```
Comment (rnyak) [inline 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: there is a typo: `chech`.  fix it as `check`.

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md [hunk 5]
```diff
@@ -11,8 +23,11 @@ DATA_ROOT_PATH=~/transformers4rec_paper_preproc_datasets_public
 3) Go to the project root path: `cd Transformers4Rec/`
 4) Create a `conda` environment and install the library dependencies, according to the instructions [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/recsys2021/hf4rec)
 4) Run the command of one of the following experiment groups (dataset, algorithm) to reproduce paper results. P.s. The reported numbers in the paper are the average of the metrics of 5 runs with different random seeds (`--seed`).
+5) When you run the commands, it will be generated within `--output_dir` CSV files with the metrics for each evaluation time window and also the Average Over Time (AOT) metrics reported in the paper.
```
Comment (rnyak) [inline 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: this sentence does not read well. How about that one instead:

5) When you run the commands, CSV files with the metrics for each evaluation time window and also the Average Over Time (AOT) metrics reported in the paper will be generated within `--output_dir`.

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md [hunk 6]
```diff
@@ -11,8 +23,11 @@ DATA_ROOT_PATH=~/transformers4rec_paper_preproc_datasets_public
...
+5) When you run the commands, it will be generated within `--output_dir` CSV files with the metrics for each evaluation time window and also the Average Over Time (AOT) metrics reported in the paper.  
+6) You can chech the metrics plots with Tensorboard (logs saved in the output dir) and with [Weights & Biases](http://wandb.ai/) service (with a free account). For that you need to install wandb (`pip install wandb`) and run `wandb login` to provide your authentication key. The run stats and plots will be saved to the `huggingface` project on W&B service by default.
 
-P.s. All neural-based models were trained for the paper experiments using a single GV100 GPU with 32 GB RAM, except the baseline Session k-NN models (V-SkNN, STAN and VSTAN) which use only CPUs.
+## Hardware environment
+All neural-based models were trained for the paper experiments using a single GV100 GPU with 32 GB RAM, except the baseline Session k-NN models (V-SkNN, STAN and VSTAN) which use only CPUs.
```
Comment (rnyak) [inline 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: I'd rephrase this sentence like that:


"All neural-based models trained on a single GV100 GPU with 32 GB RAM, except the baseline Session k-NN models (V-SkNN, STAN and VSTAN) trained only on CPUs."

**Review by rnyak — APPROVED** [review]

File: 2021_acm_recsys_transformers4rec/README.md (hunk 1 above)
Comment (gabrielspmoreira) [reply 2021_acm_recsys_transformers4rec/README.md:30]: You are right. The new script for paper reproducibility is available in [Transformers4Rec repo](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples/t4rec_paper_experiments). But the command lines for each experiment group will remain in the publications repo. I have removed the above block and included the updated content in this [markdown doc](https://github.com/NVIDIA-Merlin/publications/blob/t4r_paper_repro/2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md) of publications repo.

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md (hunk 2 above)
Comment (gabrielspmoreira) [reply 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: Fixed

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md (hunk 3 above)
Comment (gabrielspmoreira) [reply 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: Done

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md (hunk 4 above)
Comment (gabrielspmoreira) [reply 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: Done

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md (hunk 5 above)
Comment (gabrielspmoreira) [reply 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: Much better! Thanks

File: 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md (hunk 6 above)
Comment (gabrielspmoreira) [reply 2021_acm_recsys_transformers4rec/experiments_reproducibility_commands.md]: Perfect. Thanks!
